protocol/xmrmaker/claim.go

```diff
@@ -91,50 +95,65 @@ func (s *swapState) claimFunds() (ethcommon.Hash, error) {
 		)
 	}
 
-	return txHash, nil
+	return receipt, nil
 }
 
-// discoverRelayersAndClaim discovers available relayers on the network,
-func (s *swapState) discoverRelayersAndClaim() (ethcommon.Hash, error) {
-	relayers, err := s.Backend.DiscoverRelayers()
-	if err != nil {
-		return ethcommon.Hash{}, err
-	}
+// relayClaimWithXMRTaker relays the claim to the swap's XMR taker, who should
+// process the claim even if they are not relaying claims for everyone.
+func (s *swapState) relayClaimWithXMRTaker(request *message.RelayClaimRequest) (*ethtypes.Receipt, error) {
+	// only requests to the XMR taker set the offerID field
+	request.OfferID = &s.offer.ID
+	defer func() { request.OfferID = nil }()
 
-	if len(relayers) == 0 {
-		return ethcommon.Hash{}, errors.New("no relayers found to submit claim to")
-	}
-	log.Debugf("Found %d relayers to submit claim to", len(relayers))
-
-	forwarderAddress, err := s.Contract().TrustedForwarder(&bind.CallOpts{Context: s.ctx})
+	response, err := s.Backend.SubmitClaimToRelayer(s.info.PeerID, request)
```

Comment:
Here is one of the two places that the PeerID in the info is used.

Reply:
it's not necessary to store the peer ID here - the network is able to get it based on the offer ID and the stored swap info. I suppose that might make the network interface more complex though, since you'd need a function that submits the request based on peer ID, and one that submits it based on offer ID.